projects/plugins/jetpack/_inc/lib/class.media-extractor.php

```diff
@@ -543,14 +549,21 @@ public static function get_images_from_html( $html, $images_already_extracted, $
 			}
 
 			if ( ! in_array( $queryless, $image_list, true ) ) {
```

Comment:
Noting that this check will probably work less often now since `$image_list` will store strings less often.

Reply:
That's a good point. I added another test (and updated one) to make this a bit more explicit. In the case that an image is used once in a gallery and also simply in the content, but with different sizes or alt_text, I think it makes sense to include both. In the case that all the fields are the same, then we should not duplicate. It turns out that we were not uniquing links either. I found a TODO in the code, which I did.